src/postings/block_segment_postings.rs

```diff
@@ -274,13 +293,15 @@ impl BlockSegmentPostings {
 
     pub(crate) fn load_block(&mut self) {
         let offset = self.skip_reader.byte_offset();
-        if self.loaded_offset == offset {
-            return;
-        }
+        // I'm not sure why it's there, but it breaks with blocks that can be encoded on zero byte.
+        // if self.loaded_offset == offset {
```

Comment:
the idea is simply to avoid decompressing stuff again if we are hitting the same block.

Is this something that cannot happen, because of what's happening at a higher level?

Reply:
If not, this could actually be a big regression on some queries
(e.g. query interseciton of two terms that are strongly correlated. e.g. `Barrack AND Obama`

Reply:
I believe it could happen. I changed how tracking of whether the block is tracked is done so it doesn't fail with zero-sized blocks